clientv3/balancer.go

```diff
@@ -354,3 +363,47 @@ func getHost(ep string) string {
 	}
 	return url.Host
 }
+
+func (b *simpleBalancer) switchEndpoint() {
+	tNow := time.Now()
+	b.mu.RLock()
+
+	addrs := make([]grpc.Address, 0, len(b.addrs))
+	lastSwitchTime := b.lastSwitchTime
+
+	b.mu.RUnlock()
+
+	if lastSwitchTime.Add(30 * time.Second).After(tNow) {
```

Comment:
maybe a more better way it to choose endpoint both random and retry-count based.
for example, we put a 100 points on every endpoint, evey time the switch happens reduce one count.

and the switch choose is to select by the weight probability, if there are 5 memeber. their points is like this:
```
etcd-0: 100
etcd-1: 80
etcd-2: 60
```